functions.photon.php

```diff
@@ -114,6 +114,16 @@ function jetpack_photon_url( $image_url, $args = array(), $scheme = null ) {
 		|| $image_url_parts['host'] === jetpack_photon_parse_url( $custom_photon_url, PHP_URL_HOST )
 		|| $is_wpcom_image
 	) {
+		/*
+		 * VideoPress Poster images should only keep one param, ssl.
+		 */
+		if (
+			is_array( $args )
+			&& 'videos.files.wordpress.com' === strtolower( $image_url_parts['host'] )
+		) {
+			$args = array_intersect_key( array( 'ssl' => 1 ), $args );
```

Comment:
I may not be parsing this correctly, but this feels weird to force-add `ssl` param when we explicitly unset it earlier on if it's a file from wpcom. https://github.com/Automattic/jetpack/pull/12764/files#diff-89084afd23172bfa8218f1cf6c216b8dR97

Reply:
This was confusing for me at first, but it seems to be needed in this scenario, as per https://github.com/Automattic/jetpack/blob/3ed3de86a58730377bbc29f8b7df7aa7729f28ce/modules/videopress/utility-functions.php#L630

That said, the thumbnails seem to keep working when I remove that parameter, like so :

```diff
diff --git a/functions.photon.php b/functions.photon.php
index f3e03fe24..f7b0078d9 100644
--- a/functions.photon.php
+++ b/functions.photon.php
@@ -115,16 +115,17 @@ function jetpack_photon_url( $image_url, $args = array(), $scheme = null ) {
 		|| $is_wpcom_image
 	) {
 		/*
-		 * VideoPress Poster images should only keep one param, ssl.
+		 * VideoPress Poster images should not have any param.
 		 */
 		if (
 			is_array( $args )
 			&& 'videos.files.wordpress.com' === strtolower( $image_url_parts['host'] )
 		) {
-			$args = array_intersect_key( array( 'ssl' => 1 ), $args );
+			$photon_url = $image_url;
+		} else {
+			$photon_url = add_query_arg( $args, $image_url );
 		}
 
-		$photon_url = add_query_arg( $args, $image_url );
 		return jetpack_photon_url_scheme( $photon_url, $scheme );
 	}
 
diff --git a/modules/videopress/utility-functions.php b/modules/videopress/utility-functions.php
index 9bdb9a735..62374341d 100644
--- a/modules/videopress/utility-functions.php
+++ b/modules/videopress/utility-functions.php
@@ -627,8 +627,7 @@ function video_image_url_by_guid( $guid, $format ) {
 
 	$meta = wp_get_attachment_metadata( $post->ID );
 
-	// We add ssl => 1 to make sure that the videos.files.wordpress.com domain is parsed as photon.
-	$poster = apply_filters( 'jetpack_photon_url', $meta['videopress']['poster'], array( 'ssl' => 1 ), 'https' );
+	$poster = apply_filters( 'jetpack_photon_url', $meta['videopress']['poster'], array(), 'https' );
 
 	return $poster;
 }

```

I'm not quite sure what to make of it. I am afraid I am missing something that this parameter is supposed to be doing.

Unfortunately there is not a lot of info from when this was originally introduced:
https://github.com/Automattic/jetpack/pull/6631/commits/5848c2eaece91cffd585e921a5a55c0baf18b7ab

Should I attempt without that parameter?

Reply:
Very confusing, but IPF since it's only doing this for video image URLs anyway.